Clarify wording around `nix-shell --run` (#1027)
NixOS/nix.dev: source/tutorials/first-steps/ad-hoc-shell-environments.md

There are three things going on here:

> 1. `--run` executes the given command within the resulting Nix shell (which is a [Bash](https://www.gnu.org/software/bash/) sub-shell) and exits when it's done.

Reviewer: ```suggestion
1. `--run` executes the given [Bash command](https://www.gnu.org/software/bash/manual/bash.html#Shell-Commands) within the environment created by Nix, and exits when it's done.
```

More of a source of confusion would be that there really isn't such thing as a "Nix shell" - all Nix does is create an environment to spawn Bash with. There's a page in Bash documentation that expands on the concept of a command. But other than the statement is correct: That flag takes a Bash command.

Author: Wow, I always struggled to understand the concept of "Nix shell" and now I learned something new! I think adding your comment into a collapsible `<detail>` section would be enlightening. Thank you!

Reviewer: I don't feel comfortable with explaining away a misconception that was built up somewhere else. The idea here is to provide a good approximation to the truth, compactly. Ideally nix.dev is the first encounter for Nix users, and with what we give them they will be able to decipher the rest of the ecosystem.

Author: Understandable. This closed PR will be here if someone else falls into the same trap as me. Thanks again!
